12-kubernetes-cluster.tf

```diff
@@ -114,24 +112,32 @@ resource "azurerm_kubernetes_cluster" "kubernetes_cluster" {
 
 resource "azurerm_role_assignment" "genesis_managed_identity_operator" {
   principal_id         = azurerm_kubernetes_cluster.kubernetes_cluster.kubelet_identity[0].object_id
-  scope                = data.azurerm_user_assigned_identity.aks.id
+  scope                = data.azurerm_user_assigned_identity.aks[0].id
   role_definition_name = "Managed Identity Operator"
+
+  count = var.kubelet_identity_enabled ? 1 : 0
 }
 
 resource "azurerm_role_assignment" "uami_rg_identity_operator" {
   principal_id         = azurerm_kubernetes_cluster.kubernetes_cluster.kubelet_identity[0].object_id
   scope                = data.azurerm_resource_group.managed-identity-operator.id
   role_definition_name = "Managed Identity Operator"
+
+  count = var.kubelet_identity_enabled ? 1 : 0
```

Comment:
can't we just use the existing feature flag? `kubelet_uami_enabled` ? it's for the same thing